ledger/acctonline.go

```diff
@@ -364,12 +368,25 @@ func (ao *onlineAccounts) prepareCommit(dcc *deferredCommitContext) error {
 	// being updated multiple times. When that happen, we can safely omit the intermediate updates.
 	dcc.compactOnlineAccountDeltas = makeCompactOnlineAccountDeltas(deltas, dcc.oldBase, lruAccounts{})
 
-	ao.accountsMu.RUnlock()
-
 	dcc.onlineAccountExpiredRowids = expirations
 	dcc.expirationOffset = expirationOffset
 
-	dcc.genesisProto = ao.ledger.GenesisProto()
+	dcc.genesisProto = config.Consensus[ao.ledger.GenesisProto()]
+
+	start, err := ao.roundParamsOffset(dcc.oldBase)
+	if err != nil {
+		return err
+	}
+	end, err := ao.roundParamsOffset(dcc.newBase)
+	if err != nil {
+		return err
+	}
+	dcc.onlineRoundParams = ao.onlineRoundParamsData[start:end]
+	maxOnlineLookback := basics.Round(ao.maxOnlineLookback() - int(offset))
+	dcc.maxLookbackRound = basics.Round(0)
+	if ao.latest() > maxOnlineLookback {
```

Comment:
this logic should be shared (perhaps with a helper function that can be unit tested) for expiring both the new onlineaccounts table and the onlineroundparamstail